ctx.go

```diff
@@ -370,10 +371,60 @@ func (c *DefaultCtx) Response() *fasthttp.Response {
 	return &c.fasthttp.Response
 }
 
+type Fmt struct {
+	MediaType string
+	Handler   func(Ctx) error
+}
+
 // Format performs content-negotiation on the Accept HTTP header.
+// It uses Accepts to select a proper format and calls the matching
+// user-provided handler function.
+// If no accepted format is found, and a format with MediaType "default" is given,
+// that default handler is called. If no format is found and no default is given,
+// StatusNotAcceptable is sent.
+func (c *DefaultCtx) Format(handlers ...Fmt) error {
+	if len(handlers) == 0 {
+		panic("Format requires at least one handler")
```

Comment:
We should avoid using panic in handlers

Reply:
`errors.New` should be used right? @efectn

Reply:
> `errors.New` should be used right? @efectn

Yes create new error variable

Reply:
vote for sentinel errors

Reply:
You're right, an error (visible as a 5xx response) is better. I will do the same at the unreachable code on line 421.

Reply:
Hmm, should this condition return an error or nil?  As a user I'd probably assume no handlers means "do nothing successfully", but Express returns a 500 (a bit ungracefully).

Reply:
> Hmm, should this condition return an error or nil? As a user I'd probably assume no handlers means "do nothing successfully", but Express returns a 500 (a bit ungracefully).

Yeah i'd also return nil instead of error. What do you think @sixcolors @gaby @ReneWerner87

Reply:
would do it like express, if nothing is configured, it should be an error and it is pointless to use this methodic

Reply:
Yup, return error. Nil is too ambigious.

Reply:
having no handlers seems like a error, what is an intentional use case?